modules/apps/foundation/portal-configuration/portal-configuration-test-util/src/main/java/com/liferay/portal/configuration/test/util/ConfigurationTemporarySwapper.java

```diff
@@ -65,6 +58,13 @@ public ConfigurationTemporarySwapper(
 
 		_oldProperties = _configuration.getProperties();
 
+		_serviceServiceTracker = ServiceTrackerFactory.open(
+			bundle, serviceClass);
+
+		Object service = _serviceServiceTracker.waitForService(5000);
+
+		_validateService(serviceClass, service, pid);
+
 		_serviceListener = new ConfigurationServiceListener(
```

Comment:
@shuyangzhou the reason for getting the service was ordered before getting the configuration was to avoid creating a configuration if there was an exception thrown by the `_validateService` method.  The very act of requesting a configuration from ConfigurationAdmin will create one if it does not exist.  This is also the reason for the null check in 25973dd02e4b2166a12e3d1a0b5e1ed161ad5681.